File: Python_Codes/Searches.py

#Linear search
l1=[1,2,3,4,5]
ele=int(input("enter element u want to srch"))
flag=0
for i in range(0,len(l1)):
 print(l1[i])
 if ele in l1:
  flag=1
 else:
  flag=0
if flag==1:
 print("found")
else:
 print("not found") 
 

#Binary search

l=[2,5,8,12,16,23,38,56,72,91]
ele=int(input("enter element"))
mid=int(len(l)/2)
print(mid)
m=l[mid]
print(m)
flag=0

for i in range(0,len(l)):
 if(ele<mid):
  for j in range(0,mid-1):
   if ele in l:
    flag=1
  
 elif(ele>mid):
  for k in range(len(l),mid+1,-1):
   if ele in l:
    flag=1
  
 elif(ele==mid):
   flag=1
   
 else:
  flag=0 
if flag==1:
 print("found")
else:
 print("not found") 

File: Data_Structure_Codes_in_C++/Searches.cpp
#include<iostream>
using namespace std;
class search
{
    public:
    int a[10];
    int no,n,i,j;
    void read()
    {
        cout<<"\n how many?";
        cin>>n;
        cout<<"\n enter elements";
        for(i=0;i<n;i++)
        {
            cin>>a[i];
        }
        cout<<"\n elements are:";
        for(j=0;j<n;j++)
        {
            cout<<a[j];
        }
    }
    void linear()
    {
        int flag=0;
        cout<<"\n enter no do u want to srch";
        cin>>no;
        for(i=0;i<n;i++)
        {
            if(a[i]==no)
            {
                flag=1;
                break;
            }
            else
            {
                flag=0;
            }
        }
        if(flag==1)
        {
            cout<<"\n element is present:"<<a[i];
        }
        else
        {
            cout<<"\n element is not present";
        }
    }
    void binary()
    {
        int high=n-1;
        int low=0;
        int mid,flag=0;
        cout<<"\n enter elemnt do u want to srch";
        cin>>no;
        while(high>=low && flag==0)
        {
            mid=(high+low)/2;
            if(no==a[mid])
            {
                flag=1;
                break;
            }
            else if(no<a[mid])
            {
                high=mid-1;
                
            }
            else if(no>a[mid])
            {
                low=mid+1;
               
            }
        }
        if(flag==1)
        {
            cout<<"\n found";
        }
        else
        {
            cout<<"\n not found";
        }
    }
};
int main()
{
    search s;
    int ch;
    char c;
    do
    {
        cout<<"\n enter ur choice: \n 1.read \n 2.linear search \n 3.binary search";
        cin>>ch;
        switch(ch)
        {
            case 1:
              s.read();
              break;
            case 2:
              s.linear();
              break;
            case 3:
              s.binary();
              break;
        }
        cout<<"\n do u want to continue?";
        cin>>c;
    }while(c=='y');
}